### ISSUE TITLE
Enhancement - A resource pool change should not force a new TF resource

### ISSUE THREAD

**Issue by jason-azze** [terraform-providers/terraform-provider-vsphere#247]
### Terraform Version

Terraform v0.10.8

### vSphere Provider Version

0.4.2

### Affected Resource(s)

Please list the resources as a list, for example:
- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
resource "vsphere_virtual_machine" "generic" {
  name         = "${var.name_prefix}${format("%02d", count.index + var.name_starting_val)}"
  count        = "${var.how_many}"
  folder       = "${var.vm_folder_name}"
...
  # The trick is to include the word Resources in the path.
  resource_pool = "${var.vm_cluster_name}/Resources/${var.resource_pool_designator}"
...
}
```
where `resource_pool_designator` is 
```
variable "resource_pool_designator" {
  description = "The name or path for the resource pool in which to create the resources, not including the datacenter name or Resources. e.g. AllResources/DevCIl"
}
```

### Desired Behavior

Moving to a resource pool should be a "change". (If you did this in the vSphere Web Client, you could multiselect the VMs and migrate them with zero downtime in about ten seconds.)

### Actual Behavior
All VMs are slated for destruction and recreation, which is an expensive action for such a minor change. I think this is because a resource pool change triggers an `id` change.
(The output below has been edited for length and enhanced/contrived for dramatic effect.) :-)
```
myhost$ terraform plan -var-file="jenkins-agents.tfvars" -out=resource-pools-2

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

vsphere_virtual_machine.generic[0]: Refreshing state... (ID: DevCI/cent69-build20)
vsphere_virtual_machine.generic[1]: Refreshing state... (ID: DevCI/cent69-build21)

[snipped for length]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ module.linux-build.vsphere_virtual_machine.generic[0] (new resource required)
      id:                                     "DevCI/cent69-build20" => <computed> (forces new resource)
      annotation:                             "some notes here" => ""
      cluster:                                "BladeRunner01" => "BladeRunner01"
[snip]
      network_interface.0.subnet_mask:        "" => <computed>
      power_state:                            "poweredOn" => "poweredOn"
      resource_pool:                          "BladeRunner01/Resources/" => "BladeRunner01/Resources/AllResources/DevCI" (forces new resource)
      skip_customization:                     "false" => "false"
      time_zone:                              "Etc/UTC" => "Etc/UTC"
      uuid:                                   "422dc51b-6991-64c6-0d5c-6cb40ae85c4b" => <computed>
      vcpu:                                   "4" => "4"
      wait_for_customization_timeout:         "" => "10"
      wait_for_guest_net:                     "true" => "true"
[snip]

Plan: 1000 to add, 0 to change, 1000 to destroy.

------------------------------------------------------------------------
```

### Use Case or Steps to Reproduce

1. Provision 1000 VMs in resource-pool-02.
2. Decide in the morning stand-up that those 1000 VMs really belong in resource-pool-01 and say to your boss, "This shouldn't be a big deal."
3. Change one line of TF code and run a plan.
4. Say "Uh oh" when you see the plan results.

### References
Sorta...
#214

**Comment by vancluever** [comment]
Hey @jason-azze, #244 includes the ability to do vMotion on resource pools. See the [virtual machine migration](https://github.com/terraform-providers/terraform-provider-vsphere/blob/f-vm-resource-refactor/website/docs/r/virtual_machine.html.markdown#virtual-machine-migration) section.

I think the fact that resource pool changes are permitted too is an important note. I am going to review the code to make sure that it's not overlooked. It's intended to be a supported case, so if it's not, it's a bug.

Will update in #244. Will leave this open so we can close it when #244 merges.

**Comment by rmtilson** [comment]
I am confused about the vMotion support. Announcement for v1.0 says supported and your above statement says it is supported. I tried to change resource pools and it errored out for me.  Nothing fancy just the root resource pool.  

Just want to verify that this is a bug.

Terraform: 11.3
Vsphere provider: 1.3.3
vsphere: 6.5U3

```
[data.vsphere_datacenter.dc: Refreshing state...
data.vsphere_resource_pool.pool: Refreshing state...
data.vsphere_network.network: Refreshing state...
data.vsphere_datastore.datastore: Refreshing state...
vsphere_virtual_machine.test1: Refreshing state... (ID: 422bfa3c-c69c-11a5-4cb1-d8e5c6ecea7b)

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ vsphere_virtual_machine.test1
      resource_pool_id: "resgroup-65" => "resgroup-63"


Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vsphere_virtual_machine.test1: Modifying... (ID: 422bfa3c-c69c-11a5-4cb1-d8e5c6ecea7b)
  resource_pool_id: "resgroup-65" => "resgroup-63"

Error: Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.test1: 1 error(s) occurred:

* vsphere_virtual_machine.test1: error running VM migration: host ID "host-197" is not a member of resource pool "resgroup-63"]
```

**Comment by vancluever** [comment]
Hey @rmtilson, this is a bug. I've opened #424 to properly track this as this is an old issue and should have been closed when we came out with 1.0.

Thanks!
